contrib/filter-lcov.py

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env python3
+
+import argparse
+
+parser = argparse.ArgumentParser(description='Remove the coverage data from a tracefile for all files matching the pattern.')
+parser.add_argument('pattern', help='the pattern of files to remove')
+parser.add_argument('tracefile', help='the tracefile to remove the coverage data from')
+parser.add_argument('outfile', help='filename for the output to be written to')
+
+args = parser.parse_args()
+tracefile = args.tracefile
+pattern = args.pattern
+outfile = args.outfile
+
+in_remove = False
+with open(tracefile, 'r') as f:
+    with open(outfile, 'w') as wf:
+        for line in f:
+            if line.startswith("SF:") and pattern in line:
+                in_remove = True
```

Comment:
The format of the file is

```
SF:<path to file>
<bunch of lines>
end_of_record
<repeat>
```
In order to know that we are in a `<bunch of lines>` section that should be removed (because the `SF:...` line matches our removal pattern, we need to have a variable to remember to remember that we shouldn't be writing these lines to the output file. That is what `in_remove` is for.